File: 400-499/453/453.cpp
#include <vector>

using namespace std;

class Solution {
public:
	int minMoves(vector<int>& nums) {
		int min = INT_MAX;
		long sum = 0;
		
		for (auto n : nums) {
			// Get sum of all numbers
			sum += n;
			// Find the min number
			if (min > n)
				min = n;
		}
		
		// How many moves are needed to reduce all numbers to the value of the min number
		return sum - min * nums.size();
	}
};

File: 0400-0499/453/453.go
package main

import "math"

func minMoves(nums []int) int {
	var min int = math.MaxInt64
	var sum int = 0

	for _, n := range nums {
		sum += n
		if min > n {
			min = n
		}
	}

	return sum - min*len(nums)
}
